src/02-common-principles.md

```diff
@@ -125,6 +125,25 @@ The solutions will change from case to case and publicly available datasets will
 be reviewed to include common data types in the future releases of the BIDS
 specification.
 
+### Metadata conflicts
+
+BIDS specifies that metadata be stored in human-readable formats such as
+TSV (see [Tabular files](#tabular-files)) or
+JSON (see [Key/value files](#keyvalue-files-dictionaries)).
+In some cases, this duplicates metadata internal to the imaging file.
+
+In cases of conflict, the BIDS metadata is considered authoritative.
+If BIDS metadata is defined,
+format-specific metadata MUST NOT conflict to the extent permitted by the format.
+
+If an exact representation of the BIDS metadata is possible in the format,
+then the metadata SHOULD be duplicated inside the imaging file.
+If an exact representation of BIDS metadata is not possible in the format,
+then the imaging file metadata SHOULD be undefined
+or set to the closest possible approximation of the BIDS metadata.
```

Comment:
```suggestion
If only possibly an exact representation of BIDS metadata is possible in the format,
then the data file metadata SHOULD be set to the closest possible approximation of the BIDS metadata.
```
I think would make it less ambiguous.  I do not think we should allow for completely undefined in such cases.  If "representation" is not possible at all  -- there is nothing to talk about.

Reply:
The situation I was considering here is that someone might prefer to zero out a value rather than have an approximate but incorrect value in the header. For instance, suppose you have a slice timing sequence that doesn't fit any defined `NIFTI_SLICE_SEQ_*` field. I'd rather set it `NIFTI_SLICE_UNKNOWN` than the closest approximation.

I suspect this type of case is going to be relatively common for binary formats.

Reply:
I see. Yes, for `NIFTI1_SLICE_ORDER` indeed should be set to `NIFTI_SLICE_UNKNOWN` and thus indeed "undefined".

Majority of formats likely to not have `nil` or `None` or explicit "unknown" like in your example to specify the "undefined" value. So might be worth explicitly stating to set to "undefined" if explicit value is reserved for that (i.e. not `0` for some int or float field)?  and also make it as a measure of the last resort while still keeping "SHOULD" on approximate values to make those files usable with generic tools?

Reply:
What do you think of 679b8e0?

Reply:
ping @yarikoptic